shared/naturalcrit/markdown.js

```diff
@@ -13,6 +13,63 @@ renderer.html = function (html) {
 	return html;
 };
 
+// Mustache-style Divs {{class \n content ... \n}}
+let blockCount = 0;
+renderer.paragraph = function(text){
+	const blockReg = /^\s*{{[\w|,]*$|^\s*}}$/gm;
+	const inlineFullReg = /{{[^\n]*}}/g;
+	const inlineReg = /{{[\w|,]*|}}/g;
+	let matches;
+
+	//DIV - BLOCK-LEVEL
+	if(matches = text.match(blockReg)) {
+		let matchIndex = 0;
+		const res =  _.reduce(text.split(blockReg), (r, text)=>{
+
+			if(text) r.push(Markdown(text, { renderer: renderer }));
+
+			const block = matches[matchIndex] ? matches[matchIndex].trimLeft() : '';
+			if(block && block[0] == '{') {
+				r.push(`\n\n<div class="block ${block.substring(2).split(',').join(' ')}">`);
+				blockCount++;
+			} else if(block == '}}' && blockCount !== 0){
+				r.push('</div>\n\n');
+				blockCount--;
+			}
+
+			matchIndex++;
+
+			return r;
+		}, []).join('\n');
+		return res;
+	} else if(matches = text.match(inlineFullReg)) {
+
+		//SPAN - INLINE
+		matches = text.match(inlineReg);
+		let matchIndex = 0;
+		const res =  _.reduce(text.split(inlineReg), (r, text)=>{
+
+			if(text) r.push(Markdown.parseInline(text, { renderer: renderer }));
+
+			const block = matches[matchIndex] ? matches[matchIndex].trimLeft() : '';
+			if(block && block[0] == '{') {
+				r.push(`<span class="inline-block ${block.substring(2).split(',').join(' ')}">`);
+				blockCount++;
+			} else if(block == '}}' && blockCount !== 0){
+				r.push('</span>');
+				blockCount--;
+			}
+
+			matchIndex++;
+
+			return r;
+		}, []).join('');
+		return `\n<p>${res}</p>\n`;
```

Comment:
Why do we wrap inline span into a paragraph? Paragraph turns it into a block element instead of leaving it as inline one

Reply:
This isn't wrapping the span in a paragraph. What the above code does, is investigate an existing `<p>` tag, search inside for any `{{` or `}}`, and replace with `<span>` or `</span>` as appropriate. Then, with the replacements done, it returns the modified paragraph contents, which we have to re-wrap in a new `<p>` tag to preserve the original paragraph. Does that make sense? Otherwise any `<p>contents contents {{tag tag}}</p>` with {{ inside will lose its `<p>` tag.

Although... I have just realized this means it does not work inside of certain contexts, such as within a list. The bullet point text is not counted as a paragraph.... Maybe I need to make it search *any* text instead of just paragraphs. Hrm...
```
* bullet
* bullet
* bullet with {{this span}}
```

Reply:
This whole section has been totally rewritten and works much better now.